src/backend/distributed/utils/background_jobs.c

```diff
@@ -385,292 +648,344 @@ CitusBackgroundTaskQueueMonitorMain(Datum arg)
 
 	ereport(DEBUG1, (errmsg("started citus background task queue monitor")));
 
-	MemoryContext perTaskContext = AllocSetContextCreate(CurrentMemoryContext,
-														 "PerTaskContext",
-														 ALLOCSET_DEFAULT_MINSIZE,
-														 ALLOCSET_DEFAULT_INITSIZE,
-														 ALLOCSET_DEFAULT_MAXSIZE);
-
 	/*
 	 * First we find all jobs that are running, we need to check if they are still running
 	 * if not reset their state back to scheduled.
 	 */
-	{
-		StartTransactionCommand();
-		PushActiveSnapshot(GetTransactionSnapshot());
-
-		ResetRunningBackgroundTasks();
+	StartTransactionCommand();
+	PushActiveSnapshot(GetTransactionSnapshot());
 
-		PopActiveSnapshot();
-		CommitTransactionCommand();
-	}
+	ResetRunningBackgroundTasks();
 
+	PopActiveSnapshot();
+	CommitTransactionCommand();
 
-	MemoryContext oldContextPerJob = MemoryContextSwitchTo(perTaskContext);
-	TimestampTz backgroundWorkerFailedStartTime = 0;
+	/* create a map to store parallel task executors */
+	HTAB *backgroundExecutorHandles = CreateBackgroundExecutorHash(backgroundTaskContext);
```

Comment:
I recently made some nice helpers for hashtables that you can use instead now:
```suggestion
	HTAB *backgroundExecutorHandles = CreateSimpleHashWithName(int64, BackgroundExecutorHashEntry, "Background Executor Hash")`;
```